src/Query/QueryBuilder.php

```diff
@@ -309,14 +309,51 @@ public function fetchFirstColumn(): array
      */
     public function executeQuery(): Result
     {
+        [$params, $types] = $this->boundParameters();
+
         return $this->connection->executeQuery(
             $this->getSQL(),
-            $this->params,
-            $this->types,
+            $params,
+            $types,
             $this->resultCacheProfile,
         );
     }
 
+    /**
+     * Retrieve parameters and types bound to all queries (optional CTEs and main query).
+     *
+     * @return array{
+     *     list<mixed>|array<string, mixed>,
+     *     WrapperParameterTypeArray,
+     * } The parameters and types bound to the CTE queries merged with those bound to the main query.
+     */
+    private function boundParameters(): array
```

Comment:
I don't really like that the method name starts with a noun. Should it be something like `buildParametersAndTypes()`?